test/e2e/util_gatewayapi_test.go

```diff
@@ -609,30 +610,58 @@ func assertGatewaySuccessful(t *testing.T, namespace, name string) (*gatewayapiv
 
 	gw := &gatewayapiv1.Gateway{}
 	nsName := types.NamespacedName{Namespace: namespace, Name: name}
-	recordedConditionMsg := "not found"
+	recordedAcceptedConditionMsg, recordedProgrammedConditionMsg := "", ""
 
-	err := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 1*time.Minute, false, func(context context.Context) (bool, error) {
+	// Wait for the gateway to be accepted and programmed.
+	// Load balancer provisioning can take several minutes on some platforms.
```

Comment:
nit: it would be great in the future to set timeouts that vary based on platforms.

Reply:
Is there an advantage to using a lower timeout on platforms that don't need the higher timeout?  

I would prefer to take the longest amount of time we expect provisioning to take, and double it.  We don't pay the cost unless provisioning actually takes that amount of time, but we do pay a cost if the test times out when waiting longer would have allowed it to pass.

Reply:
We pay the cost of higher timeouts when everything is failing.  I'm suggesting that we be more choosy with expected timeouts to take advantage of tests failing faster on platforms that don't take as long to provision.  It's just a nit though, so no big deal.

Reply:
> We pay the cost of higher timeouts when everything is failing.

Can we test for "everything is failing" in the polling loop, and exit early if the condition is detected?  That would fail faster, flake less, and provide useful diagnostic information right in the test output.

Reply:
Yes, but let's prioritize the merge so we can get better CI signal.

Reply:
> Yes, but let's prioritize the merge so we can get better CI signal.

Agreed.  The PR has already merged, but in general when we have tests that are flaky because polling loops time out, I would argue for (1) increasing timeouts as a first pass in order to reduce flakiness and (2) refining the failure conditions as a second pass in order to fail faster and improve diagnostics.  Optimizing to reduce flakiness virtually always takes precedence over optimizing to fail fast.